### PR TITLE
Add foreman_pipeline package

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ehelms** [comment]
Is foreman_pipeline Rails 4 compatible or is there a PR to update it to Rails 4? Since the packaging here will be moving to Rails 4 SCLs very soon, I wanted to check on that aspect.

**Comment by xprazak2** [comment]
@ehelms: It is not Rails 4 compatible and there is no PR to update it yet. I guess I have work to do...

**Comment by ehelms** [comment]
@xprazak2 I'd recommend that, otherwise we'd merge this and within a few days it would break

**Comment by xprazak2** [comment]
@ehelms Thanks for warning, I will be back with an update

**Comment by xprazak2** [comment]
I made changes to be ready for Rails 4, I hope I did not miss anything.

**Comment by ehelms** [comment]
@xprazak2 double checking, the changes to foreman_pipeline are rails 4 and 3 compatible or did you have to make backwards incompatible changes?

**Comment by xprazak2** [comment]
@ehelms the changes are backward compatible

File: rel-eng/tito.props [hunk 1]
```diff
@@ -18,10 +18,12 @@ whitelist = katello-repos katello-agent
 [katello-nightly-rhel6]
 disttag = .el6
 scl = tfm
+whitelist = rubygem-foreman_pipeline
 
 [katello-nightly-rhel7]
 disttag = .el7
 scl = tfm
+whitelist = rubygem-foreman_pipeline
```
Comment (ehelms) [inline rel-eng/tito.props]: These whitelists shouldn't be needed

**Comment by ehelms** [comment]
I will work on getting this merged and built over the next day or two.

**Comment by xprazak2** [comment]
Great! I removed the whitelisting from rel-eng/tito.props

**Comment by ehelms** [comment]
Can you rebase and update this PR the same as you did the previous for the scl name changes and add the whitelisting?

File: rubygem-foreman_pipeline/rubygem-foreman_pipeline.spec [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires:   %{?scl_prefix}rubygem(jenkins_api_client) < 2.0.0
+Requires:   %{?scl_prefix}rubygem(foreman_deployments) < 1.0.0
+Requires:   %{?scl_prefix}rubygem(net-ssh) >= 2.9
+Requires:   %{?scl_prefix}rubygem(net-ssh) < 3.0
+Requires:   %{?scl_prefix}rubygem(net-scp) >= 1.1
+Requires:   %{?scl_prefix}rubygem(net-scp) < 2.0
+Requires:   %{?scl_prefix}rubygem-katello >= 2.4.0
+Requires:   %{?scl_prefix}rubygem-katello < 3.0.0
```
Comment (ehelms) [inline rubygem-foreman_pipeline/rubygem-foreman_pipeline.spec:34]: Looks like this needs to be a build requires as well.

**Comment by xprazak2** [comment]
@ehelms, I added katello as build requires, but it is complaining about [missing packages](http://koji.katello.org/koji/getfile?taskID=384466&name=mock_output.log), requiring them in my spec does not seem to have an effect, any thoughts on what I am missing?

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
@xprazak2 made some progress, I think you will just need to update the spec based on the most recent error since it is erroring in the build phase. Check https://github.com/Katello/katello-packaging/blob/master/rubygem-katello/rubygem-katello.spec#L157-L158 for what you can do in yours

**Comment by ehelms** [comment]
@xprazak2 new fialure during migration but seems code specific.

**Comment by ehelms** [comment]
The current failure is because you are trying to include files in the RPM that don't exist in the gem itself. They would need to be included via https://github.com/theforeman/foreman_pipeline/blob/master/foreman_pipeline.gemspec#L16 and a new release of the gem put out. I would add the script/ directory there, but there isn't a need to deploy the Gemfile in this case since that is only used in development.

**Comment by ehelms** [comment]
Will you update tito.props to build to katello-nightly-rhel6/7 instead of thirdparty?

**Comment by xprazak2** [comment]
I updated tito.props, thanks for helping me with this @ehelms

**Comment by ehelms** [comment]
ACK, thanks @xprazak2

**Pull request merged**